src/host.rs

```diff
@@ -97,8 +97,15 @@ fn parse_address_node(node: Node) -> Result<Address, Error> {
         .attribute("addr")
         .ok_or_else(|| Error::from("expected `addr` attribute in `address` node"))?;
 
+    let vendor = node
+        .attribute("vendor")
+        .unwrap_or_default(); 
```

Comment:
I don't like this approach. If the `vendor` field is always present, we should just use `.expect()` here with the expectation that it should never fail. If the `vendor` field is option, it should be an `Option<String>`.

Reply:
The problem is that the vendor field isnt always present on nmap output..
![image](https://github.com/Ayrx/nmap_xml_parser/assets/4402040/10d9c72e-25f7-47a8-8572-e555827052ae)
Check this..
And for this i set the vendor field to an empty string if the field is not found..

what do you think about it?

Ric

Reply:
In that case, `Option<String>` is the perfect type for that.

Reply:
> In that case, `Option<String>` is the perfect type for that.

Ok thanks, i'll try to fix it.. :)